src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -239,25 +239,48 @@ public static void ValidateEntitiesDoNotGenerateDuplicateQueriesOrMutation(IDict
                 {
                     throw new DataApiBuilderException(
                         message: $"Entity {entityName} generates queries/mutation that already exist",
-                        statusCode: System.Net.HttpStatusCode.ServiceUnavailable,
+                        statusCode: HttpStatusCode.ServiceUnavailable,
                         subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
                 }
             }
         }
 
         /// <summary>
-        /// Check whether the entity name defined in runtime config only contains
-        /// characters allowed for GraphQL names.
-        /// Does not perform validation for entities which do not
+        /// Check whether the entity configuration defined in runtime config only contains characters allowed for GraphQL names
+        /// and other validations related to rest path and methods configured for the entity.
+        /// The GraphQL validation is not performed for entities which do not
         /// have GraphQL configuration: when entity.GraphQL == false or null.
         /// </summary>
         /// <seealso cref="https://spec.graphql.org/October2021/#Name"/>
         /// <param name="runtimeConfig"></param>
-        public static void ValidateEntityNamesInConfig(Dictionary<string, Entity> entityCollection)
+        public static void ValidateEntityConfiguration(Dictionary<string, Entity> entityCollection)
         {
+            // Stores the unique rest paths configured for different entities present in the config.
+            HashSet<string> restPathsForEntities = new();
+
             foreach (string entityName in entityCollection.Keys)
             {
                 Entity entity = entityCollection[entityName];
+                if (entity.Rest is not null)
+                {
+                    JsonElement restJsonElement = JsonSerializer.SerializeToElement(entity.Rest);
+
+                    // We do the validation for rest path only if the 'rest' property maps to a json object.
+                    if (restJsonElement.ValueKind is JsonValueKind.Object)
+                    {
+                        // Since 'path' is an optional property, we skip validation if its absent.
```

Comment:
Defining REST 'path' indicates that the developer doesn't want the top-level entity name to be the rest path value. Wouldn't we also need to populate `restPathsForEntities` with top-level entity names when:
(entity.REST.enabled == true AND entity.path is not specified) 
so those conflicts can be avoided?

Reply:
> Defining REST 'path' indicates that the developer doesn't want the top-level entity name to be the rest path value

Not always. It can just be set as `path: true`. But yes, valid point. We need to account for entity names as well.

Reply:
Done.

Reply:
> > Defining REST 'path' indicates that the developer doesn't want the top-level entity name to be the rest path value
> 
> Not always. It can just be set as `path: true`. But yes, valid point. We need to account for entity names as well.

Do we actually allow `path: true` ? After reviewing the config refactor PR, we double checked the runtime config schema to see that 'true' wasn't valid.

https://github.com/Azure/data-api-builder/blob/aa5c8a33bfa4213a87afc0b5a7a31dc2db0cf954/schemas/dab.draft.schema.json#L278-L317